File: LeetCode/0014-longest-common-prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int minSize = INT_MAX;
        string minStr = "";
        for(int i = 0; i<strs.size();i++){
            if(strs[i].size() < minSize){
                minSize = strs[i].size();
                minStr = strs[i];
            }
        }
        for(int i = 0; i<strs.size();i++){
            for(int j = 0; j<minStr.size();j++){
                if(minStr[j] != strs[i][j]){
                    if(j == 0){
                        minStr = "";
                    }else{
                        minStr = minStr.substr(0,j);    
                    }
                    break;
                }
            }
        }
        return minStr;
    }
};

File: LeetCode/0014-longest-common-prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let answer = strs[0];
    let aux = answer.length;
    if(strs.length == 1){
        return answer;
    }
    for(let i = 1; i < strs.length && aux > 0; i++){
        for(let j = 0; j<aux; j++){
            if(strs[i][j] != answer[j]){
                aux = j;
                answer = answer.substr(0,j);
            }
        }
    }
    return answer;
};
